pkg/gadget-service/api/api.proto

```diff
@@ -77,12 +102,64 @@ message InfoResponse {
   bool experimental = 3;
 }
 
+message GadgetData {
+  string node = 1;
+  uint32 seq = 2;
+  repeated bytes payload = 3;
+}
+
+message Param {
+  string key = 1;
+  string description = 2;
+  string defaultValue = 3;
+  string typeHint = 4;
+  string title = 5;
+  string alias = 6;
+  repeated string tags = 7;
+  string valueHint = 8;
+  repeated string possibleValues = 9;
+  bool isMandatory = 10;
+}
+
+message GadgetInfo {
+  string name = 1;
+  string url = 2;
+  repeated DataSource dataSources = 4;
+  map<string, string> annotations = 5;
+  bytes metadata = 6;
+  repeated Param params = 7;
+}
+
+message DataSource {
+  uint32 dataSourceID = 1;
+  string name = 2;
+  uint32 type = 3;
+  repeated Field fields = 4;
+  repeated string tags = 5;
+  map<string, string> annotations = 6;
+}
+
+message Field {
+  string name = 1;
+  uint32 index = 2;
+  uint32 payloadIndex = 3;
+  uint32 offs = 4;
+  uint32 size = 5;
+  uint32 flags = 6;
+  uint32 type = 7;
+  repeated string tags = 8;
+  map<string, string> annotations = 9;
+  uint32 parent = 10;
+}
+
 message GetGadgetInfoRequest {
   // params are the gadget's parameters
   map<string, string> params = 1;
 
   // args are all parameters which were not specified with a flag
   repeated string args = 2;
+
+  uint32 version = 3;
```

Comment:
Documentation about the accepted values of this field?

Reply:
I moved this into a completely new API (well, expect from the GadgetEvent) and also added a version field to the `GadgetRunRequest` to have proper versioning from the beginning.